marimbabot_planning/src/planning.cpp

```diff
@@ -110,7 +110,21 @@ moveit::planning_interface::MoveGroupInterface::Plan Planning::plan_to_mallet_po
 
     // Add link on plane constraint to ik_options
     ik_options.goals.emplace_back(new bio_ik::LinkFunctionGoal("ur5_wrist_1_link", link_on_plane_constraint));
-    
+
+    // @TODO : add quaternion constraints similar to plane to keep wrist_2_link in specific orientation
+    auto orientation_constraint = [](const tf2::Vector3& position, const tf2::Quaternion& orientation) -> double
+    {
+        tf2::Quaternion desired_orientation;
+        desired_orientation.setRPY(0.52, 0.0, 0.0); // Set roll, pitch, and yaw angles
+
+        // Calculate the angular distance between the current and desired orientations
+        tf2::Quaternion orientation_error = desired_orientation.inverse() * orientation;
+        double angular_distance = 1.0 - orientation_error.dot(orientation_error);
+        return std::pow(angular_distance, 2);
+    };
+
+    ik_options.goals.emplace_back(new bio_ik::LinkFunctionGoal("ur5_wrist_2_link", orientation_constraint));
```

Comment:
Why did you use the LinkFunctionalGoal, which is in cartesian space instead of a normal JointGoal for the over actuated wrist joint?

Reply:
@berkgungor